kernel/src/client/default/filesystem.rs

```diff
@@ -107,11 +107,19 @@ impl<E: TaskExecutor> FileSystemClient for ObjectStoreFileSystemClient<E> {
                     };
                     let store = store.clone();
                     async move {
-                        if let Some(rng) = range {
-                            store.get_range(&path, rng).await
-                        } else {
-                            let result = store.get(&path).await?;
-                            result.bytes().await
+                        match url.scheme() {
+                            "http" | "https" => {
+                                // have to annotate type here or rustc can't figure it out
+                                Ok::<bytes::Bytes, Error>(reqwest::get(url).await?.bytes().await?)
+                            }
+                            _ => {
```

Comment:
maybe should actually have s3:// or file:// etc. here and then do `_` for an error case?

Reply:
we'll end up having to cover every format that object_store supports. i think it's better to just let it go to the object_store and have that error on invalid  urls.